src/simplification/SimplifyItems.ts

```diff
@@ -1,67 +1,72 @@
 
 import { CommonModel } from "../models/CommonModel";
 import { Schema } from "../models/Schema";
-import Simplifier from "./Simplifier";
+import { Simplifier } from "./Simplifier";
 
-type output = {newModels: CommonModel[] | undefined; items: CommonModel | undefined};
+type Output = {newModels: CommonModel[] | undefined; items: CommonModel | undefined};
 /**
  * Find the enums for a simplified version of a schema
  * 
  * @param schema to find the simplified enums for
  */
-export default function simplifyItems(schema: Schema | boolean, simplifier : Simplifier) : output {
-  let commonItems : CommonModel | undefined;
-  let models : CommonModel[] | undefined;
-  if(typeof schema !== "boolean"){
-    const addToModels = (model : CommonModel[] = []) => { models = [...(models || []), ...model]; }
-    const mergeWithItem = (model : CommonModel) => {
-      commonItems = CommonModel.mergeCommonModels(commonItems, model, schema);
-    }
-    const addToItemsAndModels = (out: output) => {
-      if(out.newModels !== undefined){
-          addToModels(out.newModels);
+export default function simplifyItems(input: Schema | boolean, simplifier : Simplifier) : Output {
```

Comment:
I understand why you make these callback in every function 😸 but it's not good for optimalization. When we'll have a lot of models in input, then it could be a problem. Maybe we should make callback outside the main function and then pass as one of argument `seenSchemas`, something like:

```ts
function callback(schema: Schema | boolean, simplifier : Simplifier, Map<any, Output> = new Map()) {
  ...logic
}

export default function simplifyItems(input: Schema | boolean, simplifier : Simplifier) : Output {
  const seenSchemas: Map<any, Output> = new Map();
  return callback(input, simplifier, seenSchemas);
}
```

We only compile one time `callback` function and it saves a lot of memory/cpu.

We should do it here and in other places with `callback` function. I don't know if it should be done in this PR or in the next one, but in my opinion we should optimize it sooner or later - better sooner 😄

Reply:
Lets do that yea! would also make it more clear and it should be split up more as the code evolve probably 👍

Reply:
Why not remove the function completely and change it from: 

```ts
export default function simplifyItems(input: Schema | boolean, simplifier : Simplifier) : Output { 
```

to

```ts
export default function simplifyItems(input: Schema | boolean, simplifier : Simplifier, seenSchemas: Map<any, Output> = new Map()) : Output { 
```

😄

Reply:
Yes, this second is better :)